packages/lean-symbols/1.0.0/custom_non_overriding_map.yml

```diff
@@ -0,0 +1 @@
+map: ↦
```

Comment:
Is this needed, or could it be generated (if not already in existence) by `generate_mapping.py`?

Reply:
That is what `generate_mapping.py` uses as input. Instead of hardcoding it into the code I made it a separate file such that it is easier to modify.

Reply:
Sorry - maybe I wasn't clear enough in my question.
As it consists of just a tiny file appearing to contain just `map: ↦`, is there a reason why it couldn't be created on the fly by `generate_mapping.py` at runtime, rather than supplying it as part of the package?
I suspect most people will just install the package and use it, without investigating `generate_mapping.py`, although it will be of interest to a few.
Sorry if this seems picky - I'm just trying to understand!

Thank you for making the other changes. I think it's ready to go otherwise.

Reply:
Well, maybe I am overengineering this. I thought that I might make more changes in future versions, but right now it's only one additional mapping. If I want additional mappings it would be quite easy to add them by just editing this file. Though there is a chance that I will not use it. As a reference for the old version, I think I added over 30 symbols and changed many other mappings. Though I now want to intentionally not change the existing mappings anymore.

If it's really a problem I can move it into `generate_mapping.py`.

Reply:
Thank you for the explanation! It's always tricky converting a personal project to something for wider use.

To keep things simple, if it doesn't make a difference, and it's not too much trouble, would you remove `custom_non_overriding_map.yml` and add a few lines to `generate_mapping.py` to create it as required, please? If it's something that needs to persist from one run to another (showing my ignorance of Python yaml coding here!), do add a check whether it exists already, to avoid overwriting it.

Reply:
Ok, wait. Just to make sure let me explain in a bit more detail. This is a configuration file for the `generate_mapping.py` program. When you start the program it downloads the lean mappings and then loads this file and injects the mappings in this file into the lean mappings, and from the combined mappings then generates the espanso mappings.

This is not a file that needs to be generated at runtime. It's a file containing additional data that configures what the final mappings will look like. Maybe I failed to explain this properly before.

The alternative is to not have a configurable file but to HARDCODE into the code that this mappings should exist. I thought that was a bit less elegant. So that is the change that I would make. Hardcode the data in the script file.

Sorry if this was all clear and I am overexplaining things.

Reply:
That's helpful. It's such a small file that I thought it could just be created when required.

I'll leave it to you - just let me know when you're ready to go.

Reply:
From my side, I'd prefer to keep the current structure of the file such that it is easier to add additional mappings (which might be shortcuts to existing mapping, that is what the one mapping in the file currently is doing. Also funnily enough that shortcut is map which is a shorthand for mapsto which maps to the symbol '↦' that denotes how a mathematical function maps an input to an output).